### PR TITLE
Adds `Aggregate` extension method

### LINKED ISSUE #55: Implement `Aggregate`
## Method

- [`Aggregate<TSource,TAccumulate,TResult>(IEnumerable<TSource>, TAccumulate, Func<TAccumulate,TSource,TAccumulate>, Func<TAccumulate,TResult>)`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.aggregate?view=netcore-3.1#System_Linq_Enumerable_Aggregate__3_System_Collections_Generic_IEnumerable___0____1_System_Func___1___0___1__System_Func___1___2__)
- [`Aggregate<TSource,TAccumulate>(IEnumerable<TSource>, TAccumulate, Func<TAccumulate,TSource,TAccumulate>)`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.aggregate?view=netcore-3.1#System_Linq_Enumerable_Aggregate__2_System_Collections_Generic_IEnumerable___0____1_System_Func___1___0___1__)
- [`Aggregate<TSource>(IEnumerable<TSource>, Func<TSource,TSource,TSource>)`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.aggregate?view=netcore-3.1#System_Linq_Enumerable_Aggregate__1_System_Collections_Generic_IEnumerable___0__System_Func___0___0___0__)

### PR DESCRIPTION
**Pull request by the-pat**
Closes #55

### CODE REVIEW

**Comment by jreina** [comment]
Is the plan to create separate methods for each overload? This doesn't cover `TSource Aggregate<TSource>(IEnumerable<TSource>, Func<TSource,TSource,TSource>)`

**Comment by the-pat** [comment]
@jreina, I think it covers it in a 💩 way.

If you check out [this test](https://github.com/ShittyLabs/fart/pull/57/files#diff-82f90370be361cf5a74de10acdec3e2d3088517a1cbdd4a3e29a0ef64c60037aR38), that case is met. But this has the awkwardness of having the define the `TAccumulate` and `TResult`.

Dart doesn't allow method overloads. The alternative to this 💩 solution is something like `aggregate2` and `aggregate3`. What are the thoughts on that? It would probably have better type safety.

**Comment by jreina** [comment]
I think having a separate method for the `TSource Aggregate<TSource>(IEnumerable<TSource>, Func<TSource,TSource,TSource>)` called `Aggregate1` is acceptable. It's up to @jakehamtexas but that's what's done in the Haskell prelude and also what DangerLinq is doing

**Comment by the-pat** [comment]
@jreina, I've separated the aggregate implementation to match the prelude style of "overloading". Thanks for the suggestion, I think this allows for better type safety when using aggregate.

File: test/aggregate_test.dart [hunk 1]
```diff
@@ -0,0 +1,223 @@
+import 'package:fart/src/exception/ArgumentNullException.dart';
+import 'package:fart/src/exception/InvalidOperationException.dart';
+import 'package:test/test.dart';
+import 'package:fart/fart.dart';
+
+void main() {
+  group('Aggregate', () {
+    group('Success Scenerios', () {
```
Comment (jreina) [inline test/aggregate_test.dart]: `Scenarios` is misspelled throughout. I'm reading it as "scene-er-ee-o" in Ned Flanders's voice
image

**Review by jreina — CHANGES_REQUESTED** [review]

File: test/aggregate_test.dart (hunk 1 above)
Comment (the-pat) [reply test/aggregate_test.dart]: Well this is embarrassing...
Homer Simpson saying "Doh!"

**Comment by the-pat** [comment]
@jreina, this is ready to review again.

**Review by jreina — APPROVED** [review]

**Pull request merged**